ixmp/core.py

```diff
@@ -1174,19 +1198,24 @@ def clone(self, model=None, scenario=None, annotation=None,
                 ' release, please use `scenario`')
             scenario = kwargs.pop('scen')
 
-        first_model_year = first_model_year or 0
+        if keep_solution and first_model_year is not None:
+            raise ValueError('Use `keep_solution=False` when cloning with '
+                             '`first_model_year`!')
+
+        if platform is not None and not keep_solution:
+            raise ValueError('Cloning across platforms is only possible '
+                             'with `keep_solution=True`!')
 
-        platform = self.platform if not platform else platform
-        model = self.model if not model else model
-        scenario = self.scenario if not scenario else scenario
+        platform = platform or self.platform
+        model = model or self.model
+        scenario = scenario or self.scenario
+        args = [platform._jobj, model, scenario, annotation, keep_solution]
+        if isyear(first_model_year, 'first_model_year'):
```

Comment:
see my comment whwere `isyear()` is defined

Reply:
done